lib/vector-core/src/event/test/size_of.rs

```diff
@@ -115,7 +115,7 @@ fn log_operation_maintains_size() {
                 Action::InsertFlat { key, value } => {
                     let new_value_sz = value.size_of();
                     let old_value_sz = log_event.get_flat(&key).map_or(0, ByteSizeOf::size_of);
-                    if !log_event.contains(&key) {
+                    if !log_event.contains(key.as_str()) {
```

Comment:
Is `&key` a `&String` here? Can we implement `Path` for `&String` to keep this (and other places) working as-is?

Reply:
It is definitely possible to implement `Path` for `&String`. I intentionally decided not to because it is almost always incorrect do that. Instead of storing a `String`, the path should be pre-parsed and that used instead. This is future work though, as there are a _lot_ of places that need to change.

Reply:
Well, in terms of "future work", I was thinking doing so (at least) now seems to _significantly_ cut down on the changes introduced in this PR, as tons of files here only have the `&key` -> `key.as_str()` diff.